YSI_Visual/y_interiors/y_interiors_impl.inc

```diff
@@ -0,0 +1,300 @@
+#include "y_interiors_decl"// decl-arations
+
+/*
+
+                                                              
+88        88                            88                    
+88        88                            88                    
+88        88                            88                    
+88aaaaaaaa88   ,adPPYba,    ,adPPYba,   88   ,d8   ,adPPYba,  
+88""""""""88  a8"     "8a  a8"     "8a  88 ,a8"    I8[    ""  
+88        88  8b       d8  8b       d8  8888[       `"Y8ba,   
+88        88  "8a,   ,a8"  "8a,   ,a8"  88`"Yba,   aa    ]8I  
+88        88   `"YbbdP"'    `"YbbdP"'   88   `Y8a  `"YbbdP"'  
+                                                              
+                                                              
+
+*/
+
+#define YSI_INTERIORS_TRUE 1
+#define YSI_INTERIORS_FALSE 0
+
+#define YSI_INTERIOR_ACTION_ENTER 0
+#define YSI_INTERIOR_ACTION_EXIT 1
+
+gYSI_InteriorLoadingTime(const playerid)
+{
+	TogglePlayerControllable(playerid,true);
+}
+
+YSI_OnPlayerNearEntrance(playerid);
+public YSI_OnPlayerNearEntrance(playerid)
+{
+	new YSI_InteriorFunctionName[100];
+	new YSI_InteriorActionid;
+    TogglePlayerControllable(playerid,false);
+	for(new i; i < MAX_INTERIORS; i++)
+	{
+		if( IsPlayerInRangeOfPoint( playerid, 1.5, gYSI_InteriorData[ i ][e_YSI_INTENTRANCE_X], \
+		gYSI_InteriorData[ i ][e_YSI_INTENTRANCE_Y], gYSI_InteriorData[ i ][e_YSI_INTENTRANCE_Z]) \
+		&& GetPlayerVirtualWorld( playerid ) == gYSI_InteriorData[ i ][e_YSI_INTERIOR_VWID_OUT] \
+		&& !IsPlayerInAnyVehicle( playerid ))
+		{
+			YSI_InteriorActionid = 0;
+			format(YSI_InteriorFunctionName,sizeof(YSI_InteriorFunctionName),
+				"%s@_yI",gYSI_InteriorData[ i ][e_YSI_INTERIOR_NAME]);
+

```

Comment:
The commits pushed have `\r\n`, if you're using git please make sure you only push `\n`, especially if you add a `.gitattributes` saying as much.